### DISCUSSION TITLE
[SC]: Servizio getNotificheByEvento

### DISCUSSION THREAD

**Discussion by siscomanpr** [italia/ansc#831] [Miglioramenti] [answered]
### Ambiente

pre-produzione

### Servizio

R008 - Servizi cooperativi notifiche

### idOperazione

2178995

### Provenienza segnalazione

Software House

### Comune di riferimento e operatore

Serravalle Sesia

### Data del test (yyyy-mm-dd)

18/04/2024

### Descrivi il problema

Buongiorno,
utilizzando il metodo getNotificheByEvento del servizio R008, il modelNotifica che viene restituito ci risulta diverso rispetto a quanto indicato nella documentazione, in particolare a quanto indicato in R008_notifiche.pdf.
Ci sono alcuni campi in più, per esempio: datamodificastato, descsoggettointestatario, flagnotificacondivisa ecc...
Potete aggiornare la documentazione e chiarire il significato/utilità di questi campi extra?

Questo è un esempio di quanto riceviamo come risposta:
{
            "flagNonLetto": null,
            "id": 2316862,
            "idansc": "2024-1312926-00000-002137",
            "idcomuneformazione": 9419,
            "idcomunecompetente": 9419,
            "idsoggettorif": 2305785,
            "datainiziovalidita": "2024-04-17T09:00:01.000+00:00",
            "datafinevalidita": "9999-12-31T09:00:01.000+00:00",
            "idusecase": "10000000",
            "descrizione": "Annotazione automatica id 2024-1312926-00000-002137 ( da inviare via PEC ) ",
            "idtipoevento": 1,
            "idtipocontenuto": 4,
            "idevento": 1312926,
            "nomecomuneform": "SERRAVALLE SESIA",
            "descsoggettointestatario": "XXXXXXXXXXX XXXXXXXXXX",
            "flageventocartaceo": 1,
            "ideventosorgente": 1312923,
            "idstatoflussonotifiche": 1,
            "idgenerenotifiche": 1,
            "datamodificastato": null,
            "flagnotificacondivisa": 0,
            "idcanalenotifica": 1
}

Grazie
Mauro Maccagno
Siscom SPA

**Comment by amusarra** [comment]
Salve @siscomanpr 
Abbiamo preso in carico la segnalazione.

Cordiali Saluti.
Supporto ANSC.

**Comment by mttfranci** [comment, marked as answer]
@siscomanpr con il rilascio della patch [1.27.6 ](https://github.com/italia/ansc/releases/tag/v1.27.6)abbiamo introdotto un meccanismo di retrocompatibilità con cui potete attivare la risposta corretta.

Dovrebbe essere sufficiente passare al path segnalato il parametro [revision](https://github.com/italia/ansc/blob/main/docs/Revision.md)=2000003, ad esempio : 

`/notifiche/getNotificheByEvento/1?revision=2000003`

In tal caso l'output restituito dovrebbe essere del tipo : 

```
        {
            "descrizione": "Dichiarazione entro i 10 giorni - resa dal padre",
            "idcanalenotifica": "0",
            "dataInizioValidita": "2024-04-04T14:17:00",
            "id": "579852",
            "idStatoFlussoNotifiche": "1",
            "idGenereNotifiche": "2",
            "idComuneFormazione": "1000000",
            "idAnsc": "2024-580261-43234-999999",
            "idComuneCompetente": "2780",
            "idSoggettoRif": "579835"
        }
```

Ovvero aderente alle specifiche openapi del servizio [R008](https://github.com/italia/ansc/blob/main/docs/openapi/R008_notifiche.yaml)

Trovate qualche dettaglio in più nel documento <https://github.com/italia/ansc/blob/main/docs/Revision.md>
Siamo a disposizione per ogni ulteriore chiarimento.

Saluti

Assistenza ANSC

**Comment by gcuci** [comment]
Buongiorno,

segnalo che il metodo di getNotificheByEvento, con l'aggiunta della revision=2000003, al path, risponde effettivamente come indicato (per un ulteriore riscontro, potete far riferimento all'operazione che abbiamo effettuato oggi, 2024-05-30, con idOperazione: "2262611").
Ma come si vede anche dal vostro esempio, manca la valorizzazione della proprietà 'dataFineValidita', prevista nel ModelNotifica.

In attesa di riscontro,

cordiali saluti.

**Reply by mttfranci** [reply]
Salve @gcuci  le notifiche non prevedono al momento una data fine validità (fino ad un certo punto in preproduzione la avevano).
Per questo il campo non viene restituito nel caso specifico.

Saluti
